src/main/java/net/dries007/tfc/api/recipes/LoomRecipe.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.api.recipes;
+
+import net.dries007.tfc.api.registries.TFCRegistries;
+import net.minecraft.item.Item;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+import javax.annotation.Nullable;
+
+public class LoomRecipe extends IForgeRegistryEntry.Impl<LoomRecipe>
+{
+    @Nullable
+    public static LoomRecipe get(Item item)
+    {
+        return TFCRegistries.LOOM.getValuesCollection().stream().filter(x -> x.isValidInput(item)).findFirst().orElse(null);
+    }
+
+    @Nullable
+    public static LoomRecipe getByOutput(Item item)
+    {
+        return TFCRegistries.LOOM.getValuesCollection().stream().filter(x -> x.isValidOutput(item)).findFirst().orElse(null);
+    }
+
+    private Item inputItem;
+    private int inputAmount;
+    private Item outputItem;
+    private int stepCount;
+    private ResourceLocation inProgressTexture;
+
+    public LoomRecipe(ResourceLocation name, Item input, int inputAmount, Item output, int stepsRequired, ResourceLocation inProgressTexture)
+    {
+        inputItem = input;
```

Comment:
Minor nitpick - It's nice to have the local variables match the constructor ones,
```suggestion
        this.inputItem = inputItem;
```